Assets/HoloToolkit/Build/Editor/BuildDeployWindow.cs

```diff
@@ -0,0 +1,581 @@
+                }
+                catch (Exception ex)
+                {
+                    Debug.LogError(ex.ToString());
+                }
+            }
+            EditorUtility.ClearProgressBar();
+        }
+
+        private bool InstallApp(string buildPath, string appName, string targetDevice)
+        {
+            // Get the appx path
+            FileInfo[] files = (new DirectoryInfo(buildPath)).GetFiles("*.appx");
+            if (files.Length == 0)
+            {
+                Debug.LogError("No APPX found in folder build folder (" + buildPath + ")");
+                return false;
+            }
+
+            // Connection info
+            var connectInfo = new BuildDeployPortal.ConnectInfo(targetDevice, deviceUser, devicePassword);
+
+            // Kick off the install
+            Debug.Log("Installing build on: " + targetDevice);
+            return BuildDeployPortal.InstallApp(files[0].FullName, connectInfo);
+        }
+
+        private void UninstallAppOnDevicesList(string appName, string[] targetList)
+        {
+            try
+            {
+                string[] IPlist = ParseIPList(targetIPs);
+                for (int i = 0; i < IPlist.Length; i++)
+                {
+                    string IP = FinalizeIP(IPlist[i]);
+                    EditorUtility.DisplayProgressBar("Uninstalling application", "Uninstall (" + IP + ")", (float)i / (float)IPlist.Length);
+                    UninstallApp(appName, IP);
+                }
+            }
+            catch (Exception ex)
+            {
+                Debug.LogError(ex.ToString());
+            }
+            EditorUtility.ClearProgressBar();
+        }
+
+        private bool UninstallApp(string appName, string targetDevice)
+        {
+            // Connection info
+            var connectInfo = new BuildDeployPortal.ConnectInfo(targetDevice, deviceUser, devicePassword);
+
+            // Kick off the install
+            Debug.Log("Uninstall build: " + targetDevice);
+            return BuildDeployPortal.UninstallApp(appName, connectInfo);
+        }
+
+        private void UpdateBuilds()
+        {
+            this.builds.Clear();
+
+            try
+            {
+                List<string> appPackageDirectories = new List<string>();
+                string[] buildList = Directory.GetDirectories(BuildDirectory_FullPath);
+                foreach (string appBuild in buildList)
+                {
+                    string appPackageDirectory = appBuild + @"\AppPackages";
+                    if (Directory.Exists(appPackageDirectory))
+                    {
+                        appPackageDirectories.AddRange(Directory.GetDirectories(appPackageDirectory));
+                    }
+                }
+                IEnumerable<string> selectedDirectories = from string directory in appPackageDirectories orderby Directory.GetLastWriteTime(directory) descending select Path.GetFullPath(directory);
```

Comment:
Nit: Didn't see any coding style guides, but this would probably be a lot more readable with some line breaks (or converted to Method syntax).